kratos/sources/kratos_application.cpp

```diff
@@ -116,7 +116,16 @@ KratosApplication::KratosApplication(const std::string& ApplicationName)
       mpConditions(KratosComponents<Condition>::pGetComponents()),
       mpModelers(KratosComponents<Modeler>::pGetComponents()),
       mpRegisteredObjects(&(Serializer::GetRegisteredObjects())),
-      mpRegisteredObjectsName(&(Serializer::GetRegisteredObjectsName())) {}
+      mpRegisteredObjectsName(&(Serializer::GetRegisteredObjectsName())) {
+        
+        Registry::SetCurrentSource(mApplicationName);
+
+        for (auto component : {"geometries", "elements", "conditions", "constraints", "modelers", "constitutive_laws"}) {
+            if (!Registry::HasItem(std::string(component))) {
+                Registry::AddItem<RegistryItem>(std::string(component)+"."+mApplicationName);
+            }
+        }
+      }
```

Comment:
Do we need this? I think, the registry will create all the parent keys if they are not present ryt? Is this done for consistency?